core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java

```diff
@@ -370,6 +371,7 @@ protected void initPlugins() {
         camelContextExtension.lazyAddContextPlugin(AnnotationBasedProcessorFactory.class,
                 this::createAnnotationBasedProcessorFactory);
         camelContextExtension.lazyAddContextPlugin(DumpRoutesStrategy.class, this::createDumpRoutesStrategy);
+        camelContextExtension.lazyAddContextPlugin(JolokiaHttpRequestHandlerFactory.class, this::createJolokiaHttpRequestHandlerFactory);
```

Comment:
Jolokia should only be discovered if enabled, eg that jolokiaEnabled option in camel-main